src/commands/deploy.js

```diff
@@ -103,9 +103,7 @@ class DeployCommand extends Command {
 
     // TODO: abstract settings lookup
     let deployFolder
-    if (flags['dir']) {
-      deployFolder = path.resolve(process.cwd(), flags['dir'])
-    } else if (get(config, 'build.publish')) {
+    if (get(config, 'build.publish')) {
       deployFolder = path.resolve(site.root, get(config, 'build.publish'))
```

Comment:
```suggestion
      deployFolder = get(config, 'build.publish')
```

`config.build.publish` is always an absolute path.